components/board/fake/board.go

```diff
@@ -286,6 +287,7 @@ func (a *Analog) reset(pin string) {
 func (a *Analog) Read(ctx context.Context, extra map[string]interface{}) (int, error) {
 	a.Mu.RLock()
 	defer a.Mu.RUnlock()
+	a.Set(rand.Intn(100))
```

Comment:
With analogs, I'm able to actually change `a.Value` to be a random number. This seems like the best way to set it, although maybe there's a better _place_ to set it.

Reply:
:shrug: This seems like as good a place to do it as any. Change the comment on line 299: `Set` has been used in much more than testing for a while.